Update C++ compiler flags for Java/Android bindings

**Describe the problem**
C++20 features like `std::strong_ordering` were introduced in the code, but the compiler flags for Java and Android binds are still using `--std=c++17` flags, which makes the builds fail for both platforms.

**To Reproduce**
Steps to reproduce the behavior:
1. Try building for Java with by following https://book.verovio.org/installing-or-building-from-sources/other-bindings.html
```bash
$ cd bindings/java
$ mvn package

[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.rism.verovio:VerovioToolkit >-------------------
[INFO] Building VerovioToolkit 4.5.0-dev
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ VerovioToolkit ---
[INFO] skip non existing resourceDirectory /Users/withsang/Develop/verovio/bindings/java/src/main/resources
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ VerovioToolkit ---
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 11 source files with javac [debug target 1.8] to target/classes
[WARNING] bootstrap class path not set in conjunction with -source 8
[INFO] 
[INFO] --- antrun:3.1.0:run (build-native) @ VerovioToolkit ---
[INFO] Executing tasks
[INFO]      [exec] In file included from ../../src/accid.cpp:8:
[INFO]      [exec] In file included from ../../include/vrv/accid.h:13:
[INFO]      [exec] In file included from ../../include/vrv/layerelement.h:17:
[INFO]      [exec] ../../include/vrv/fraction.h:46:10: error: no type named 'strong_ordering' in namespace 'std'
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |     ~~~~~^
[INFO]      [exec] ../../include/vrv/fraction.h:46:34: warning: '<=>' is a single token in C++20; add a space to avoid a change in behavior [-Wc++20-compat]
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |                                  ^ 
[INFO]      [exec]       |                                     
[INFO]      [exec] ../../include/vrv/fraction.h:46:26: error: 'operator<=' cannot be the name of a variable or data member
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |                          ^
[INFO]      [exec] ../../include/vrv/fraction.h:46:36: error: expected ';' at end of declaration list
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |                                    ^
[INFO]      [exec]       |                                    ;
[INFO]      [exec] ../../src/accid.cpp:166:29: error: no member named 'contains' in 'std::set<vrv::Accid *>'
[INFO]      [exec]   166 |         if (!adjustedAccids.contains(accid)) return;
[INFO]      [exec]       |              ~~~~~~~~~~~~~~ ^
[INFO]      [exec] 1 warning and 4 errors generated.
[INFO]      [exec] In file included from ../../src/adjustaccidxfunctor.cpp:8:
[INFO]      [exec] In file included from ../../include/vrv/adjustaccidxfunctor.h:11:
[INFO]      [exec] In file included from ../../include/vrv/functor.h:11:
[INFO]      [exec] In file included from ../../include/vrv/comparison.h:11:
[INFO]      [exec] In file included from ../../include/vrv/artic.h:14:
[INFO]      [exec] In file included from ../../include/vrv/layerelement.h:17:
[INFO]      [exec] ../../include/vrv/fraction.h:46:10: error: no type named 'strong_ordering' in namespace 'std'
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |     ~~~~~^
[INFO]      [exec] ../../include/vrv/fraction.h:46:34: warning: '<=>' is a single token in C++20; add a space to avoid a change in behavior [-Wc++20-compat]
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |                                  ^ 
[INFO]      [exec]       |                                     
[INFO]      [exec] ../../include/vrv/fraction.h:46:26: error: 'operator<=' cannot be the name of a variable or data member
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |                          ^
[INFO]      [exec] ../../include/vrv/fraction.h:46:36: error: expected ';' at end of declaration list
[INFO]      [exec]    46 |     std::strong_ordering operator<=>(const Fraction &other) const;
[INFO]      [exec]       |                                    ^
[INFO]      [exec]       |                                    ;
[INFO]      [exec] ../../src/adjustaccidxfunctor.cpp:62:30: error: no member named 'contains' in 'std::set<vrv::Accid *>'
[INFO]      [exec]    62 |         if (m_adjustedAccids.contains(accid)) continue;
[INFO]      [exec]       |             ~~~~~~~~~~~~~~~~ ^
[INFO]      [exec] ../../src/adjustaccidxfunctor.cpp:105:31: error: no member named 'contains' in 'std::set<vrv::Accid *>'
[INFO]      [exec]   105 |         if (!m_adjustedAccids.contains(accids.at(i))) {
[INFO]      [exec]       |              ~~~~~~~~~~~~~~~~ ^
[INFO]      [exec] ../../src/adjustaccidxfunctor.cpp:113:31: error: no member named 'contains' in 'std::set<vrv::Accid *>'
[INFO]      [exec]   113 |         if (!m_adjustedAccids.contains(accids.at(j))) {
[INFO]      [exec]       |              ~~~~~~~~~~~~~~~~ ^
[INFO]      [exec] 1 warning and 6 errors generated.

(And so on)
```


**Expected behavior**
A clear and concise description of what you expected to happen.

The build successes and the example code successfully generates `output.svg`.

**Input data**
The smallest example (MEI unless about another input format) to reproduce the problem

None

**Verovio information**
 - Version of Verovio [e.g. develop, 3.4.0]: develop
 - Binding [e.g. command-line, JavaScript, Python]: Java

**Environment information (as appropriate)**
 - OS: [e.g. MacOS, Window]: macOS 14.7.1 23H222 arm64, Apple clang version 16.0.0 
 - Browser [e.g. Chrome, Safari]: None

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/bindings/android/jni/Application.mk
+++ b/bindings/android/jni/Application.mk
@@ -1,6 +1,6 @@
 # only build armeabi-v7a, if required add 'armeabi' or 'x86'
 APP_ABI := armeabi-v7a
-APP_CPPFLAGS += -fexceptions -frtti -std=c++17
+APP_CPPFLAGS += -fexceptions -frtti -std=c++20
 APP_OPTIM := release
 APP_STL := c++_shared
 APP_MODULES := verovio-android

--- a/bindings/java/build.sh
+++ b/bindings/java/build.sh
@@ -24,7 +24,7 @@ FILES="$SRCFILES \
  ../../src/json/jsonxx.cc \
  ../../src/crc/crc.cpp"
 
-CXXOPTS="-g -fpic -std=c++17 -I../../include -I../../include/vrv -I../../include/json -I../../include/hum -I../../include/crc -I../../include/midi -I../../include/pugi -I../../include/zip -I../../libmei/addons -I../../libmei/dist -I/opt/local/include/ "
+CXXOPTS="-g -fpic -std=c++20 -I../../include -I../../include/vrv -I../../include/json -I../../include/hum -I../../include/crc -I../../include/midi -I../../include/pugi -I../../include/zip -I../../libmei/addons -I../../libmei/dist -I/opt/local/include/ "
 
 PATHS=""
 unamestr=$(uname)

